### DISCUSSION TITLE
Root component tag

### DISCUSSION THREAD

**Discussion by taublast** [hydrostack/hydro#56] [General]
Should we introduce an auto-generated root component tag like <component></component> to make coding more relaxed, in order to avoid errors when no root tag is set by the developer?

`InvalidOperationException: The wire component must have only one root element.` will be thrown in the following cases:

Example A:

```
@if(Flag)
{
   //no error
   <div>component output</div>
}
else
{
   //nothing here, and no root tag throws an error!
}
```

Example B, error:

```
 <div>blabla</div>
 <div>blablabla</div>
```

Would You agree Hydro framework should auto-generate its root tag for components to avoid this error?

**Comment by kjeske** [comment]
I was thinking about it in an early stage, but it seemed inconsistent and implicit, to sometimes create the tag, and sometimes not, depending on the view content. It could be an unexpected behaviour to add such default tag behind the scenes. Also choosing such tag wouldn't be obvious. In some cases `div` is fine, but in others not. I think it's better to make the same rule for all the views, so users don't get confused.
